schema/README.md

```diff
@@ -0,0 +1,95 @@
+
+Today most of the Observability solutions (splunk, datadog, dynatrace) recommend using a consolidated schema to represent the entire variance of log/trace/metrics producers.
+
+This allows monitoring, incidents investigation and corrections process to become simpler, maintainable and reproducible.
+
+A Schema-Aware visualization component is a component which assumes the existence of specific index/indices name patterns and expects these indices to have a specific structure - a schema.
+
+As an example we can see that Trace-Analytics is schema-aware since it directly assumes the traces & serviceMap indices exist and expects them to follow a specific schema.
+
+This definition doesn’t change the existing status of visualization components which are not “Schema Aware” but it only regulates which Visual components would benefit using a schema and which will be agnostic of its content.
+
+Operation Panel for example, are not “schema aware” since they don’t assume in advanced the existence of a specific index nor do they expect the index they display to have a specific structure.
+
+Schema aware visualizations such as Applications, Metrics, Alerts and Integrations will not be able to work directly with a non-standard proprietary index unless being explicitly mapped during the query execution (this capability is planned to be supported in the future).
+
+## Data Model
+
+Simple Schema for Observability needs to allow ingestion of both (OTEL/ECS) formats and internally consolidate them to best of its capabilities for presenting a unified Observability platform.
+
+The data model is highly coupled with the visual components, for example - the Application visual component & Trace analytics are directly coupled with all the Observability schemas (Logs, Traces, Spans).
+
+## Observability index naming
+
+The Observability indices would follow the recommended immutable data stream ingestion pattern using the [data_stream concepts](https://opensearch.org/docs/latest/opensearch/data-streams/)
+
+Index pattern will follow the next naming structure `{type}`-`{dataset}`-`{namespace}`
+
+ - **type**	- indicated	the observability high level types "logs", "metrics", "traces", "alerts"
+ - **dataset**	- The field can contain anything that classify the source of the data - such as `nginx.access` (If none specified "**default** " will be used).
+ - **namespace**	- A user defined namespace. Mainly useful to allow grouping of data such as production grade, geography classification
+
+This strategy allows the two degrees of naming freedom: dataset and namespace. For example a customer may want to route the nginx logs from two geographical areas into two different indices:
+
+ - `logs-nginx-us`
+ - `logs-nginx-eu`
+
+This type of distinction also allows for creation of crosscutting queries by setting the next index query pattern `logs-nginx-*` or by using a geographic based crosscutting query `logs-*-eu`.
+
+## Data index routing
```

Comment:
would that be handled by metrics schema?

Reply:
the additional `sso_` prefix should theoretically prevent naming conflicts - but to your point, if customers have already created such format they will need to update our mapping template accordingly so that there will not be any conflicts prior to creation of the indices - this is why I've removed the default index creation from the plugin loading phase and it should be now explicitly invoked by the user - so that he can verify all the index configurations are correct